app/models/ext_management_system.rb

```diff
@@ -448,29 +448,45 @@ def self.destroy_queue(ids)
   end
 
   def destroy_queue
-    _log.info("Queuing destroy of #{self.class.name} with id: #{id}")
+    msg = "Queuing destroy of #{self.class.name} with id: #{id}"
+
+    _log.info(msg)
+    task = MiqTask.create(
+      :name    => "Destroying #{self.class.name} with id: #{id}",
+      :state   => MiqTask::STATE_QUEUED,
+      :status  => MiqTask::STATUS_OK,
+      :message => msg,
+    )
+
     child_managers.each(&:destroy_queue)
-    self.class.schedule_destroy_queue(id)
+    self.class.schedule_destroy_queue(id, task.id)
+
+    task.id
   end
 
-  def self.schedule_destroy_queue(id, deliver_on = nil)
+  def self.schedule_destroy_queue(id, task_id, deliver_on = nil)
     MiqQueue.put(
       :class_name  => name,
       :instance_id => id,
       :method_name => "orchestrate_destroy",
       :deliver_on  => deliver_on,
+      :args        => [task_id],
     )
   end
 
   # Wait until all associated workers are dead to destroy this ems
-  def orchestrate_destroy
+  def orchestrate_destroy(task_id)
     disable! if enabled?
 
     if self.destroy == false
       _log.info("Cannot destroy #{self.class.name} with id: #{id}, workers still in progress. Requeuing destroy...")
       self.class.schedule_destroy_queue(id, 15.seconds.from_now)
```

Comment:
Doesn't this have to change since you changed the signature of schedule_destroy_queue ?

Reply:
Yes, done